client/components/data/query-site-comments/index.jsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { requestCommentsList } from 'state/comments/actions';
+
+const requestComments = ( { requestSiteComments, siteId, status } ) => requestSiteComments( { siteId, status } );
+
+export class QuerySiteComments extends Component {
+	componentDidMount() {
+		requestComments( this.props );
+	}
+
+	componentDidUpdate( { siteId } ) {
+		if ( siteId !== this.props.siteId ) {
+			requestComments( this.props );
+		}
+	}
+
+	render() {
+		return null;
+	}
+}
+
+const mapDispatchToProps = dispatch => ( {
+	requestSiteComments: ( { siteId, status = 'unapproved' } ) => dispatch( requestCommentsList( {
```

Comment:
Because this is how our default comment management will be as if the latest word from @drw158 and because approved by default would be much riskier from the perspective of accidentally loading a million things